float32[] cube_pos
float32[] ball_pos
float32[] ball_vel